bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/ContentFragmentUtils.java

```diff
@@ -235,13 +236,23 @@ public static String getGridResourceType(Resource fragmentResource) {
     }
 
     /**
-     * Returns a map of all given resources mapping their name to their {@link ComponentExporter component exporter}
-     * model.
+     * Gets a map of all resources with the resource name as the key and the corresponding {@link ComponentExporter}
+     * model as the value.
+     *
+     * The returned map is ordered as provided by the resourceIterator.
+     * Any resource that cannot be adapted to a {@link ComponentExporter} model is omitted from the returned map.
+     *
+     * @param resourceIterator Iterator of resources for which to get the component exporters.
+     * @param modelFactory Model factory service.
+     * @param slingHttpServletRequest Current request.
+     * @return Ordered map of resource names to {@link ComponentExporter} models.
      */
-    public static Map<String, ComponentExporter> getComponentExporters(Iterator<Resource> resourceIterator,
-                                                                       ModelFactory modelFactory,
-                                                                       SlingHttpServletRequest slingHttpServletRequest) {
-        final Map<String, ComponentExporter> componentExporterMap = new LinkedHashMap<>();
+    @NotNull
+    public static LinkedHashMap<String, ComponentExporter> getComponentExporters(
```

Comment:
I'd keep the return type as it was before: `Map<String, ComponentExporter>`.
This way we keep the same pattern for the `ExperienceFragment` model:
- API: `Map<String, ? extends ComponentExporter> getExportedItems()`
- implementation: we return a `LinkedHashMap`

Reply:
Since this class is internal, I see no reason not to change to a better contract (`LinkedHashMap` ensures order)

Reply:
sure we can the more precise return type `LinkedHashMap `. My suggestion was more to have consistent return types.